aiida/backends/sqlalchemy/migrations/versions/e797afa09270_reset_hash.py

```diff
@@ -40,7 +40,7 @@ def upgrade():
     conn = op.get_bind()
 
     # Invalidate all the hashes & inform the user
-    echo_warning('Invalidating all the hashes of all the nodes. Please run verdi rehash', bold=True)
+    echo.echo_info('Invalidating the hashes of all nodes. Please run verdi rehash on existing data.', bold=True)
```

Comment:
I think that if we really think this message is important for users, we need to keep it a warning otherwise it is for sure getting ignored. If that is fine, then we should just remove it. Alternatively, to prevent it from being shown unnecessarily (i.e. when creating a new database) we should just add a simply check on the count of nodes. Only show the message if the node table is not empty.

Reply:
> we should just add a simply check on the count of nodes. Only show the message if the node table is not empty.

That would be perfect - I thought of this initially but didn't know how to do this in a migration.
Can you point me to an example that I can adapt to achieve this?
I guess I need to use some low-level interface to check whether there are any nodes.

Reply:
You could replace the `notify_user` function with something like the following:
```
def notify_user(apps, schema_editor):  # pylint: disable=unused-argument
    DbNode = apps.get_model('db', 'DbNode')
    if DbNode.objects.count():
        echo_warning('Invalidating all the hashes of all the nodes. Please run verdi rehash', bold=True)
```
In SqlAlchemy you will have to do a manual SQL query, something like `SELECT count(*) FROM db_dbnode;`